src/js/lib/helpers.js

```diff
@@ -290,4 +291,78 @@
 		}
 	};
 
+	/**
+	 * Finds the root domain
+	 */
+	object.findRootDomain = function () {
+		var cookiePrefix = '_sp_root_domain_test_';
+		var cookieName = cookiePrefix + new Date().getTime();
+		var cookieValue = '_test_value_' + new Date().getTime();
+
+		var split = window.location.hostname.split('.');
+		var position = split.length - 1;
+		while (position >= 0) {
+			var currentDomain = split.slice(position, split.length).join('.');
+			cookie.cookie(cookieName, cookieValue, 0, '/', currentDomain);
```

Comment:
I think it's slightly confusing to use 0 as the ttl here - it relies on the fact that browser-cookie-lite interprets a ttl of 0 as meaning that no ttl is set rather than that the ttl should be 0 seconds. Could we change it to null?